src/tasks.js

```diff
@@ -0,0 +1,117 @@
+ko.tasks = (function () {
+    var scheduler,
+        taskQueue = [],
+        taskQueueLength = 0,
+        handleOrigin = 0,
+        processingTask,
+        observer;
+
+    if (observer = window['MutationObserver'] || window['WebKitMutationObserver']) {
+        // Chrome 18+, Firefox 14+, IE 11+, Opera 15+, Safari 6+; borrowed from https://github.com/petkaantonov/bluebird
+        scheduler = (function (callback) {
+            var div = document.createElement("div");
+            new observer(callback).observe(div, {attributes: true});
+            return function () { div.classList.toggle("foo"); };
+        })(scheduledProcess);
+    } else if (document && "onreadystatechange" in document.createElement("script")) {
+        // IE 6�10; borrowed from https://github.com/YuzuJS/setImmediate
+        scheduler = function (callback) {
+            var script = document.createElement("script");
+            script.onreadystatechange = function () {
+                script.onreadystatechange = null;
+                document.documentElement.removeChild(script);
+                script = null;
+                callback();
+            };
+            document.documentElement.appendChild(script);
+        };
+    } else {
+        scheduler = function (callback) {
+            setTimeout(callback, 0);
```

Comment:
It may make sense to have a generic `ko.utils.runAsync(fn)` that uses animation frames by default, falls back to `setTimeout`, and is overloadable for testing. This would be used elsewhere, too.